src/ocl/convolutionocl.cpp

```diff
@@ -671,28 +671,28 @@ static void DirConvFindCore(Handle& handle,
         EvaluateInvokers(handle, all, algorithm_name, network_config, invoke_ctx, record);
     }
 
-    // Direct algo
     if(!use_winograd_only)
     {
-        ConvolutionUserBuffers bufs(workSpace, workSpaceSize);
-        bufs.SetFwd(x, w, y);
-        const auto all = conv.FindDataDirectSolutions(
-            handle, xDesc, wDesc, yDesc, exhaustiveSearch, true, bufs, invoke_ctx);
-        PrecompileSolutions(handle, all);
-        const auto algorithm_name = AlgorithmName{"miopenConvolutionFwdAlgoDirect"};
-        EvaluateInvokers(handle, all, algorithm_name, network_config, invoke_ctx, record);
-    }
-
-    // Implicit GEMM algo
-    if(!use_winograd_only)
-    {
-        ConvolutionUserBuffers bufs(workSpace, workSpaceSize);
-        bufs.SetFwd(x, w, y);
-        const auto all = conv.FindDataImplicitGemmSolutions(
-            handle, xDesc, wDesc, yDesc, exhaustiveSearch, true, bufs, invoke_ctx);
-        PrecompileSolutions(handle, all);
-        const auto algorithm_name = AlgorithmName{"miopenConvolutionFwdAlgoImplicitGEMM"};
-        EvaluateInvokers(handle, all, algorithm_name, network_config, invoke_ctx, record);
+        // Direct algo
+        {
+            ConvolutionUserBuffers bufs(workSpace, workSpaceSize);
+            bufs.SetFwd(x, w, y);
+            const auto all = conv.FindDataDirectSolutions(
+                handle, xDesc, wDesc, yDesc, exhaustiveSearch, true, bufs, invoke_ctx);
+            PrecompileSolutions(handle, all);
+            const auto algorithm_name = AlgorithmName{"miopenConvolutionFwdAlgoDirect"};
+            EvaluateInvokers(handle, all, algorithm_name, network_config, invoke_ctx, record);
+        }
+        // Implicit GEMM algo
```

Comment:
Does cppcheck 2.2 require this change? If yes, then please disable related check and roll this back.

Reply:
This check was previously disabled due to too many false positives. Cppcheck 2.2 no longer has FPs in miopen, so I re-enabled it. This is not a FP. There is no point in checking `use_winograd_only` twice when a an unamed scope should be used.

Reply:
There is a point. I do not want extra indenting, so I combined scope with `if`. Do you insist?

Reply:
[Recommendation]

Reply:
An unnamed scope makes the intention clearer. If you want to avoid indentation and code duplication then a local lambda could be used:

```cpp
    auto eval = [&](auto find, auto name) {
        ConvolutionUserBuffers bufs(workSpace, workSpaceSize);
        bufs.SetFwd(x, w, y);
        const auto all = find(
            conv, handle, xDesc, wDesc, yDesc, exhaustiveSearch, true, bufs, invoke_ctx);
        PrecompileSolutions(handle, all);
        const auto algorithm_name = AlgorithmName{name};
        EvaluateInvokers(handle, all, algorithm_name, network_config, invoke_ctx, record);
    };

    if(!use_winograd_only)
    {
        // Direct algo
        eval(std::mem_fn(&ConvolutionDescriptor::FindDataDirectSolutions), "miopenConvolutionFwdAlgoDirect");
        // Implicit GEMM algo
        eval(std::mem_fn(&ConvolutionDescriptor::FindDataImplicitGemmSolutions), "miopenConvolutionFwdAlgoImplicitGEMM");
    }
```

Reply:
Thanks for the advice, Paul.